show/main.py

```diff
@@ -1087,6 +1086,31 @@ def table(verbose):
 # 'platform' group ("show platform ...")
 #
 
+def get_hw_info_dict():
+    """
+    This function is used to get the HW info helper function  
+    """
+    hw_info_dict = {}
+    machine_info = sonic_platform.get_machine_info()
+    platform = sonic_platform.get_platform_info(machine_info)
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    data = config_db.get_table('DEVICE_METADATA')
+    try: 
+        hwsku = data['localhost']['hwsku']
+    except KeyError:
+        hwsku = "Unknown"
+    version_info = sonic_platform.get_sonic_version_info()
+    asic_type = version_info['asic_type']
+    hw_info_dict['platform'] = platform
+    hw_info_dict['hwsku'] = hwsku
+    hw_info_dict['asic_type'] = asic_type
+    return hw_info_dict
+
+# This calls teh get_hw_info_dict function and creates a dictionary
+hw_info_dict = get_hw_info_dict()
```

Comment:
I don't think this should be a global variable, because `get_hw_info_dict()` will be called every time a `show` command is run, and this info isn't necessary for the majority of commands, so this will incur unnecessary cycle and potential slowdown of commands that don't need it. Please move this line into the functions that will use it, making `hw_info_dict` a local variable in those functions.

Reply:
updated to remove  global variable.